zerver/views/thumbnail.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse, HttpResponseForbidden
+from django.conf import settings
+from six.moves import urllib
+from zerver.models import UserProfile, validate_attachment_request
+from zerver.lib.request import has_request_variables
+from zerver.lib.thumbnail import generate_thumbnail_url
+
+def validate_thumbnail_request(user_profile, path):
+    # type: (UserProfile, str) -> bool
+    if path.startswith('user_uploads/'):
```

Comment:
So one thing that worries me about these checks is that I'm never sure whether there's a leading `/` on `path` without reading a lot of code.  Maybe just deserves a comment here.

Also, we should update the mypy annotations to the Python 3 style.